### DISCUSSION TITLE
How to close app after every test , after upgrading to 1.22.2 it doesnt happen anymore

### DISCUSSION THREAD

**Discussion by Hamzastreamwide** [appium/appium#16382] [Q&A]
Hi , si i upgraded appium to 1.22.2 , to have the fix of crash in alertview for ios , the old behavior of closing automatically the app after each test , and relaunching the session again before the test , is not working any more  :( , so i want someone to help me with this why delete session is not called after the test is finished

**Comment by KazuCocoa** [comment]
Could you create an issue with logs (addressed in an issue template)?

**Reply by Hamzastreamwide** [reply]
okaay i will i will put the link of issue here

**Comment by Hamzastreamwide** [comment]
https://github.com/appium/appium/issues/16396

**Comment by Hamzastreamwide** [comment]
can u explain those too , "shouldTerminateApp":false,"forceAppLaunch":false and thanks

**Reply by KazuCocoa** [reply]
Perhaps you set `noReset` capability as true. Then, `shouldTerminateApp` set as false since noReset means appium does not reset/clean the app under test.
Please read https://github.com/appium/appium-xcuitest-driver for capabilities.

**Reply by HamzaMiz** [reply]
i found solution i turned shouldterminateApp to true

**Comment by Hamzastreamwide** [comment]
capabilities.setCapability("shouldTerminateApp",true);
 okay after i added this the app is closed than relaunched
